service/src/main/java/bio/terra/landingzone/library/landingzones/management/ResourcesReader.java

```diff
@@ -11,5 +11,9 @@ public interface ResourcesReader {
 
   List<DeployedResource> listResourcesByPurpose(ResourcePurpose purpose);
 
+  List<DeployedResource> listResourcesWithPurpose();
+
   List<DeployedVNet> listVNetWithSubnetPurpose(SubnetResourcePurpose purpose);
+
+  List<DeployedVNet> listVNetResourcesWithSubnetPurpose();
```

Comment:
Just for my own edification, why the special handling for VNet/subnets?

Reply:
The VNet purpose does not make practical sense - we need to look at subnets and their purposes. Getting to subnets requires is a separate ARM API, via network().